src/onnx/parse_convolution.cpp

```diff
@@ -42,18 +43,201 @@ struct parse_convolution : op_parser<parse_convolution>
+                                          instruction_ref& unshifted_input)
+    {
+        auto unshifted_input_half = info.add_instruction(
+            migraphx::make_op("convert", {{"target_type", migraphx::shape::half_type}}),
+            unshifted_input);
+
+        auto input_shifted_half = info.add_common_op("add", unshifted_input_half, offset_op);
+
+        return info.add_instruction(
+            migraphx::make_op("convert", {{"target_type", migraphx::shape::int8_type}}),
+            input_shifted_half);
+    }
+
+    static void shift_input_and_bias(const onnx_parser::node_info& info,
+                                     const instruction_ref& offset_op,
+                                     const bool has_bias,
+                                     instruction_ref& input,
+                                     instruction_ref& input_bias)
+    {
+        input = add_int8_shift(info, offset_op, input);
+        if(has_bias)
+        {
+            input_bias = add_int8_shift(info, offset_op, input_bias);
+        }
+    }
+
+    static float get_symmetric_value(const instruction_ref& input)
+    {
+        float symmetric_value = 0;
+        // adjust symmetric zero point value for uint8 types
+        if(input->get_shape().type() == migraphx::shape::uint8_type)
+        {
+            symmetric_value = 128;
+        }
+        return symmetric_value;
+    }
+
+    static instruction_ref gen_symmetric_literal(const instruction_ref& input,
+                                                 const bool is_quant_conv,
+                                                 onnx_parser::node_info& info)
+    {
+        instruction_ref ret = input;
+        if(is_quant_conv)
+        {
+            float symmetric_value = get_symmetric_value(input);
+            ret                   = info.add_literal(migraphx::literal{
+                migraphx::shape{input->get_shape().type(), {1}, {0}}, {symmetric_value}});
+        }
+
+        return ret;
+    }
+
+    static instruction_ref get_zero_point(const instruction_ref& input,
+                                          int index,
+                                          const bool is_quant_conv,
+                                          onnx_parser::node_info& info,
+                                          const std::vector<instruction_ref>& args)
+    {
+        instruction_ref ret = input;
+        if(args.size() > index)
+        {
+            // Check for type mismatch on parse
+            if(input->get_shape().type() != args[index]->get_shape().type())
+                MIGRAPHX_THROW("PARSE:Conv Data and Data Zero Point must have same type");
+
+            ret = args[index];
+            if(is_symmetric_zero_point(ret))
+            {
+                std::cout << "Hit symmetric input" << std::endl;
+                ret = gen_symmetric_literal(ret, is_quant_conv, info);
+            }
+            ret->debug_print();
+        }
+        else
```

Comment:
Both the if/else look identical..

Reply:
The intent here is to add a valid literal that's symmetric scalar value so that other parts of the code don't get triggered generating residual convolution blocks.

Reply:
Ah I think the confusion lies in the fact this parser handles both Conv and ConvInteger inputs